doc/adrs/088_add_support_for_compartment_expand-collapse.adoc

```diff
@@ -0,0 +1,170 @@
+            switch(node.getCollapsingState()) {
+                case CollapsingState.expanded:
+                    collapsingTools.add(expandTool);
+                    break;
+                case CollapsingState.collapsed:
+                    collapsingTools.add(collapseTool);
+                    break;
+                default:
+                    break;
+            }
+            var expandCollapseToolSection = ToolSection.newToolSection("expand-collapse-section")
+                    .label("")
+                    .imageURL("")
+                    .tools(collapsingTools)
+                    .build();
+            extraToolSections.add(expandCollapseToolSection);
+        }
+    }
+}
+
+----
+
+To handle the action we will add a new concepts to out api
+
+===== Graphql API
+
+[source,graphql]
+----
+
+extend type Mutation {
+    collapseExpand(input: CollapseExpandInput!): CollapseExpandPayload!
+}
+
+input CollapseExpandInput {
+    id: ID!
+    editingContextId: ID!
+    representationId: ID!
+    nodeId: ID!
+    collapsingState: CollapsingState!
+}
+
+enum CollapsingState {
+    COLLAPSED
+    EXPANDED
+}
+
+union CollapseExpandPayload = ErrorPayload | CollapseExpandSuccessPayload
+
+type CollapseExpandSuccessPayload {
+    id: ID!
+}
+
+----
+
+===== Input and Handler
+
+
+[source,java]
+----
+
+public record CollapseExpandInput(UUID id, String editingContext, String representationId, String nodeId, CollapsingState collapsingState) implements IDiagramInput {}
+
+public class CollapseExpandEventHandler implements IDiagramEventHandler {
+
+    @Override
+    public void handle(One<IPayload> payloadSink, Many<ChangeDescription> changeDescriptionSink, IEditingContext editingContext, IDiagramContext diagramContext, IDiagramInput diagramInput) {
+        if (diagramInput instanceof CollapseExpandInput collapseExpandInput) {
+            this.handleCollapseExpand(payloadSink, changeDescriptionSink, editingContext, diagramContext, diagramInput);
+        } else {
+            // send error payload
+        }
+    }
+
+    private void handleCollapseExpand(One<IPayload> payloadSink, Many<ChangeDescription> changeDescriptionSink, IEditingContext editingContext, IDiagramContext diagramContext, CollapseExpandInput collapseExpandInput) {
+        ...
+
+        diagramContext.setDiagramEvent(new CollapseExpandEvent(collapseExpandInput.nodeId, CollapsingState.COLLAPSED.equals(collapseExpandInput.collapsingState)));
+        payloadSink.tryEmitValue(new CollapseExpandSuccessPayload(collapseExpandInput.id));
+        changeDescriptionSink.tryEmitValue(new ChangeDescription(DiagramChangeKind.DIAGRAM_ELEMENT_COLLAPSING_STATE_CHANGE, diagramInput.representationId(), diagramInput));
+    }
+
+}
+
+----
+
+== Status
+
+Proposed
+
+== Consequence
+
+The boolean `isCollapsible` on `view::NodeDescription` could become `CollapsingStateExpression` used to dynamically return the possibility for a node to be expandable, collapsible or none.
```

Comment:
and NodeDescription#isCollapsible could become NodeDescription#isCollapsiblePredicate()